heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricsmgr.sink;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import com.twitter.heron.common.basics.TypeUtils;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsInfo;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsRecord;
+import com.twitter.heron.spi.metricsmgr.sink.IMetricsSink;
+import com.twitter.heron.spi.metricsmgr.sink.SinkContext;
+
+
+/**
+ * A metrics sink that publishes metrics on a http endpoint
+ */
+public class WebSink implements IMetricsSink {
+  private static final Logger LOG = Logger.getLogger(WebSink.class.getName());
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private static final int HTTP_STATUS_OK = 200;
+
+  private static final String KEY_PORT = "port";
+  private static final String KEY_PORT_FILE = "port-file";
+  private static final String KEY_PATH = "path";
+  private static final String KEY_FLAT_METRICS = "flat-metrics";
+  private static final String KEY_INCLUDE_TOPOLOGY_NAME = "include-topology-name";
+
+  protected ConcurrentHashMap<String, Object> metrics = new ConcurrentHashMap<>();
```

Comment:
Not sure if changing your metrics names over time is a great design either ;) but I will fix that.